build-patched-kao-kcmo-images.sh

```diff
@@ -154,7 +157,11 @@ patch_and_push_image cluster-kube-apiserver-operator
 patch_and_push_image cluster-kube-controller-manager-operator
 create_new_release_with_patched_images
 
-base_image=$(grep "^FROM openshift/ose-base" crc-cluster-kube-apiserver-operator/Dockerfile | sed 's/^FROM //')
-
-update_base_image crc-dnsmasq "${base_image}"
-update_base_image crc-routes-controller "${base_image}"
+# In case there is no change in the openshift component then the base
+# image is also not changed so no need to build dnsmasq/route images
+if [ -f crc-cluster-kube-apiserver-operator/Dockerfile ]; then
+    base_image=$(grep "^FROM openshift/ose-base" crc-cluster-kube-apiserver-operator/Dockerfile | sed 's/^FROM //')
+    
+    update_base_image crc-dnsmasq "${base_image}"
+    update_base_image crc-routes-controller "${base_image}"
```

Comment:
Arguably we also need to call `skopeo copy --dest-authfile ${OPENSHIFT_PULL_SECRET_PATH} --all docker://registry-proxy.engineering.redhat.com/rh-osbs/${brew_repo}:latest docker://quay.io/crcont/${brew_repo#crc-}:${openshift_version}` even if we did not rebuild anything, but we don't use the openshift version tag for these, so we can also remember that if an openshift release is missing, this means it used the previous one.

Reply:
@cfergeau Right we should but the problem is as of now we are using `docker://registry-proxy.engineering.redhat.com/rh-osbs/${brew_repo}:latest` so if I run this script for 4.12.8 and those images never built on the brew then everything works as expected but then even by mistake someone run this script for 4.12.6 where the image already exist so the brew build not happen for KAO/KCMO and just tagging happen (which is again pushing same image on the same tag) but if we add same logic for dnsmasq/router image then latest is 4.12.8 and it will update the same for 4.12.6. Which is also OK as long as there is no base image change but might create issue if image is updated. 

What we need is same logic as we have for KAO/KCMO where we identify build and if present then don't rebuild but if not then build.